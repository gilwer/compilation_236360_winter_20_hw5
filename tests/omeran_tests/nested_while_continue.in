void main() {
    int i = 1;
    int j = 1;
    while (i < 50) {
        i = 1 + i;
        if (i/10 == (i-1)/10) {
            print("i not divisible by 10, i is");
            printi(i);
            continue;
        }
        while (j < 40) { // only happens once, j is not reset
            j = j+1;
            if (j/7 != (j-1)/7) {
                print("j divisible by 7!");
                printi(j);
            } else if (j / 5 != (j-1)/5) {
                continue;
            }
            print("shouldn't print when j is divisible by 5 but not 7");
            print("j is");
            printi(j);
            continue;
            continue;
            continue;
        }
    }
}