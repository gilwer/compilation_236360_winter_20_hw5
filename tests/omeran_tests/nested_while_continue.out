i not divisible by 10, i is
2
i not divisible by 10, i is
3
i not divisible by 10, i is
4
i not divisible by 10, i is
5
i not divisible by 10, i is
6
i not divisible by 10, i is
7
i not divisible by 10, i is
8
i not divisible by 10, i is
9
shouldn't print when j is divisible by 5 but not 7
j is
2
shouldn't print when j is divisible by 5 but not 7
j is
3
shouldn't print when j is divisible by 5 but not 7
j is
4
shouldn't print when j is divisible by 5 but not 7
j is
6
j divisible by 7!
7
shouldn't print when j is divisible by 5 but not 7
j is
7
shouldn't print when j is divisible by 5 but not 7
j is
8
shouldn't print when j is divisible by 5 but not 7
j is
9
shouldn't print when j is divisible by 5 but not 7
j is
11
shouldn't print when j is divisible by 5 but not 7
j is
12
shouldn't print when j is divisible by 5 but not 7
j is
13
j divisible by 7!
14
shouldn't print when j is divisible by 5 but not 7
j is
14
shouldn't print when j is divisible by 5 but not 7
j is
16
shouldn't print when j is divisible by 5 but not 7
j is
17
shouldn't print when j is divisible by 5 but not 7
j is
18
shouldn't print when j is divisible by 5 but not 7
j is
19
j divisible by 7!
21
shouldn't print when j is divisible by 5 but not 7
j is
21
shouldn't print when j is divisible by 5 but not 7
j is
22
shouldn't print when j is divisible by 5 but not 7
j is
23
shouldn't print when j is divisible by 5 but not 7
j is
24
shouldn't print when j is divisible by 5 but not 7
j is
26
shouldn't print when j is divisible by 5 but not 7
j is
27
j divisible by 7!
28
shouldn't print when j is divisible by 5 but not 7
j is
28
shouldn't print when j is divisible by 5 but not 7
j is
29
shouldn't print when j is divisible by 5 but not 7
j is
31
shouldn't print when j is divisible by 5 but not 7
j is
32
shouldn't print when j is divisible by 5 but not 7
j is
33
shouldn't print when j is divisible by 5 but not 7
j is
34
j divisible by 7!
35
shouldn't print when j is divisible by 5 but not 7
j is
35
shouldn't print when j is divisible by 5 but not 7
j is
36
shouldn't print when j is divisible by 5 but not 7
j is
37
shouldn't print when j is divisible by 5 but not 7
j is
38
shouldn't print when j is divisible by 5 but not 7
j is
39
i not divisible by 10, i is
11
i not divisible by 10, i is
12
i not divisible by 10, i is
13
i not divisible by 10, i is
14
i not divisible by 10, i is
15
i not divisible by 10, i is
16
i not divisible by 10, i is
17
i not divisible by 10, i is
18
i not divisible by 10, i is
19
i not divisible by 10, i is
21
i not divisible by 10, i is
22
i not divisible by 10, i is
23
i not divisible by 10, i is
24
i not divisible by 10, i is
25
i not divisible by 10, i is
26
i not divisible by 10, i is
27
i not divisible by 10, i is
28
i not divisible by 10, i is
29
i not divisible by 10, i is
31
i not divisible by 10, i is
32
i not divisible by 10, i is
33
i not divisible by 10, i is
34
i not divisible by 10, i is
35
i not divisible by 10, i is
36
i not divisible by 10, i is
37
i not divisible by 10, i is
38
i not divisible by 10, i is
39
i not divisible by 10, i is
41
i not divisible by 10, i is
42
i not divisible by 10, i is
43
i not divisible by 10, i is
44
i not divisible by 10, i is
45
i not divisible by 10, i is
46
i not divisible by 10, i is
47
i not divisible by 10, i is
48
i not divisible by 10, i is
49
